File: src/Helpers/ArgParser.java
package Helpers;

import Models.Publisher;

import java.util.ArrayList;
import java.util.List;

public class ArgParser {
    public static List<Publisher> fetchPublishersFromCommandLine(String[] args) throws IllegalArgumentException {
        List<Publisher> publishers = new ArrayList<>();
        int position = -1;
        for (int i = 0; i < args.length; i++) {
            if (args[i].equals("-p")) {
                position = i;
            }
        }
        if (position == -1)
            throw new IllegalArgumentException();
        try {
            while (args[++position] != null) {
                String[] data = args[position].split(":");
                Publisher publisher = new Publisher(data[0], Integer.parseInt(data[1]));
                publishers.add(publisher);
            }
            // This line never executes, however compiler warned me.
            throw new IndexOutOfBoundsException();
        } catch (IndexOutOfBoundsException err) {
            return publishers;
        }
    }
}

File: src/helpers/ArgParser.kt
package helpers

import models.Broker
import models.Publisher
import java.util.*

object ArgParser {
    @Throws(IllegalArgumentException::class)
    fun fetchPublishersFromCommandLine(args: Array<String>, returnType: Any): List<Any> {
        val entities = ArrayList<Any>()
        var position = -1
        for (i in args.indices) {
            if (args[i] == "-p") {
                position = i
            }
        }
        if (position == -1)
            throw IllegalArgumentException()
        try {
            while (args[++position] != null) {
                val data = args[position].split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                var entity: Any
                when (returnType) {
                    "publisher" -> entity = Publisher(data[0], Integer.parseInt(data[1]))
                    "broker" -> entity = Broker(data[0], Integer.parseInt(data[1]))
                    else -> entity = Object()
                }
                entities.add(entity)
            }
            // This line never executes, however compiler warned me.
            throw IndexOutOfBoundsException()
        } catch (err: IndexOutOfBoundsException) {
            return entities
        }

    }
}
